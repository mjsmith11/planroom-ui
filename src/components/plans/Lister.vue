<template>
    <div class="lister">
      <h4>Bidding Documents</h4>
      <ul :class="{listborder : plans.length > 0}">
        <li v-for="plan in plans" :key="plan.key"><a :href="plan.url">{{ plan.key | removeDir }}</a></li>
      </ul>
    </div>
</template>
<script>
import { EventBus } from '../../event-bus.js'
import axios from 'axios'
export default {
  data () {
    return {
      plans: []
    }
  },
  methods: {
    loadList () {
      axios.get('/jobs/' + this.jobId + '/plans')
        .then(res => {
          this.plans = res.data
        })
        // eslint-disable-next-line
        .catch(err => {
          // console.log(err)
        })
    }
  },
  props: [
    'jobId'
  ],
  filters: {
    removeDir (value) {
      var splits = value.split('/')
      splits.splice(0, 1)
      return splits.join('/')
    }
  },
  created () {
    // setTimeout(this.loadList, 1000)
    EventBus.$on('job-read', () => {
      this.$nextTick(this.loadList)
    })
    EventBus.$on('file-uploaded', this.loadList)
  }
}
</script>
<style scoped>
  .lister {
    margin-top: 25px;
    width: 100%
  }
  ul {
    width: 100%;
    padding-top: 20px;
    padding-bottom: 20px;
  }
  .listborder {
    border: 2px solid #e4e4e4;
  }
  h4 {
    font-style: italic;
    font-weight: 600;
  }
</style>
