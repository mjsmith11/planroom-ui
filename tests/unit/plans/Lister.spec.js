import { mount } from '@vue/test-utils'
import Lister from '@/components/plans/Lister'
import mockAxios from 'jest-mock-axios'
import { EventBus } from '@/event-bus.js'

describe('Lister', () => {
  let cmp
  let eventAttachSpy

  beforeEach(() => {
    cmp = mount((Lister), {
      propsData: {
        jobId: 1
      }
    })
    eventAttachSpy = jest.spyOn(EventBus, '$on')
  })

  afterEach(() => {
    // cleaning up the mess left behind the previous test
    mockAxios.reset()
    EventBus.$off('job-read')
    EventBus.$off('file-uploaded')
    jest.clearAllMocks()
  })

  it('filters object keys', () => {
    expect(cmp.vm.$options.filters.removeDir('1/myFile.txt')).toBe('myFile.txt')
  })

  it('has expected html structure', () => {
    const response = {
      data: [
        {
          key: '1/a blueprint.pdf',
          url: 'http://test.com/1/aBlueprint.com'
        },
        {
          key: '1/another blueprint.pdf',
          url: 'http://test.com/1/anotherBlueprint.com'
        }
      ]
    }
    cmp.vm.loadList()
    mockAxios.mockResponse(response)
    expect(cmp.html()).toMatchSnapshot()
  })

  it('loads the list', () => {
    const response = {
      data: [
        {
          key: '1/a blueprint.pdf',
          url: 'http://test.com/1/aBlueprint.com'
        },
        {
          key: '1/another blueprint.pdf',
          url: 'http://test.com/1/anotherBlueprint.com'
        }
      ]
    }
    cmp.vm.loadList()
    mockAxios.mockResponse(response)
    expect(cmp.vm.plans).toBe(response.data)
  })

  it('attaches events', () => {
    // two events should get attached in the create hook
    expect(eventAttachSpy).toHaveBeenCalledTimes(2)
  })
})
