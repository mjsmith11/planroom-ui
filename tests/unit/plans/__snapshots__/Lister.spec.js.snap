// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lister has expected html structure 1`] = `
<div class="lister">
  <h4>Bidding Documents</h4>
  <ul class="listborder">
    <li>
      <a href="http://test.com/1/aBlueprint.com">a blueprint.pdf</a>
    </li>
    <li>
      <a href="http://test.com/1/anotherBlueprint.com">another blueprint.pdf</a>
    </li>
  </ul>
</div>
`;
